### PR TITLE
Add Lisk Faucet Plugin - Closes #583

### LINKED ISSUE #583: Add Lisk Faucet plugin
### Description
Allow the Faucet plugin to be enabled from `lisk-core` for `devnet/betanet/testnet` networks.

- Import Faucet plugin to `src/application/index.ts` (import { FaucetPlugin } from 'lisk-sdk';)
- Update `getApplication` for registering Faucet plugin (app.registerPlugin(FaucetPlugin, { loadAsChildProcess: true });)
- Update `start.ts` to add command for enabling faucet plugin
- Add test cases `test/commands/start.spec.ts` for Faucet Plugin
- Update `readme` to reflect the relavant changes

### Acceptance Criteria

- Lisk Faucet plugin is integrated, possible to enable/disable from the command line, and docs are updated
- Make sure the faucet plugin enables only for `devnet/betanet/testnet`

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?

This PR resolves #583 

### How was it solved?

- [x] Register Faucet plugin
- [x] Add `enable-faucet-plugin` command
- [x] Add check to enable Faucet plugin only for devnet/betanet/testnet
- [x] Add test case
- [x] Update start.md 

### How was it tested?

- [x] npm install --registry https://npm.lisk.com/
- [x] npm run build
- [x] ./bin/run start -n devnet --enable-faucet-plugin

### CODE REVIEW

File: test/commands/start.spec.ts [hunk 1]
```diff
@@ -209,7 +209,7 @@ describe.skip('start', () => {
 	describe('when --api-ipc is specified', () => {
 		it('should update the config value', async () => {
 			await StartCommand.run(['--api-ipc'], config);
-			const [, usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
+			const [usedConfig,] = (application.getApplication as jest.Mock).mock.calls[0];
```
Comment (sameersubudhi) [inline test/commands/start.spec.ts:212]: Optional: the trailing comma seems to be unnecessary.
```suggestion
			const [usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
```

File: test/commands/start.spec.ts [hunk 2]
```diff
@@ -218,7 +218,7 @@ describe.skip('start', () => {
 	describe('when --api-ws is specified', () => {
 		it('should update the config value', async () => {
 			await StartCommand.run(['--api-ws'], config);
-			const [, usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
+			const [usedConfig,] = (application.getApplication as jest.Mock).mock.calls[0];
```
Comment (sameersubudhi) [inline test/commands/start.spec.ts:221]:
```suggestion
			const [usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
```

File: test/commands/start.spec.ts [hunk 3]
```diff
@@ -227,31 +227,39 @@ describe.skip('start', () => {
 	describe('when custom host with --api-ws-host is specified along with --api-ws', () => {
 		it('should update the config value', async () => {
 			await StartCommand.run(['--api-ws', '--api-ws-host', '0.0.0.0'], config);
-			const [, usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
+			const [usedConfig,] = (application.getApplication as jest.Mock).mock.calls[0];
```
Comment (sameersubudhi) [inline test/commands/start.spec.ts:230]:
```suggestion
			const [usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
```

File: test/commands/start.spec.ts [hunk 4]
```diff
@@ -227,31 +227,39 @@ describe.skip('start', () => {
...
 			expect(usedConfig.rpc.host).toBe('0.0.0.0');
 		});
 	});
 
 	describe('when custom port with --api-ws-port is specified along with --api-ws', () => {
 		it('should update the config value', async () => {
 			await StartCommand.run(['--api-ws', '--api-ws-port', '8888'], config);
-			const [, usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
+			const [usedConfig,] = (application.getApplication as jest.Mock).mock.calls[0];
```
Comment (sameersubudhi) [inline test/commands/start.spec.ts:238]:
```suggestion
			const [usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
```

File: test/commands/start.spec.ts [hunk 5]
```diff
@@ -262,15 +270,15 @@ describe.skip('start', () => {
 				['--enable-monitor-plugin', '--monitor-plugin-host', '0.0.0.0'],
 				config,
 			);
-			const [, usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
+			const [usedConfig,] = (application.getApplication as jest.Mock).mock.calls[0];
```
Comment (sameersubudhi) [inline test/commands/start.spec.ts:273]:
```suggestion
			const [usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
```

File: test/commands/start.spec.ts [hunk 6]
```diff
@@ -262,15 +270,15 @@ describe.skip('start', () => {
...
 			expect(usedConfig.plugins.monitor.host).toBe('0.0.0.0');
 		});
 	});
 
 	describe('when custom port with --monitor-plugin-port is specified along with --enable-monitor-plugin', () => {
 		it('should update the config value', async () => {
 			await StartCommand.run(['--enable-monitor-plugin', '--monitor-plugin-port', '8888'], config);
-			const [, usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
+			const [usedConfig,] = (application.getApplication as jest.Mock).mock.calls[0];
```
Comment (sameersubudhi) [inline test/commands/start.spec.ts:281]:
```suggestion
			const [usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
```

File: test/commands/start.spec.ts [hunk 7]
```diff
@@ -285,23 +293,23 @@ describe.skip('start', () => {
 				],
 				config,
 			);
-			const [, usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
+			const [usedConfig,] = (application.getApplication as jest.Mock).mock.calls[0];
```
Comment (sameersubudhi) [inline test/commands/start.spec.ts:296]:
```suggestion
			const [usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
```

File: test/commands/start.spec.ts [hunk 8]
```diff
@@ -285,23 +293,23 @@ describe.skip('start', () => {
...
 			expect(usedConfig.plugins.monitor.whiteList).toEqual(['192.08.0.1:8888', '192.08.0.2:8888']);
 		});
 	});
 
 	describe('when empty white list with --monitor-plugin-whitelist is specified along with --enable-monitor-plugin', () => {
 		it('should update the config value', async () => {
 			await StartCommand.run(['--enable-monitor-plugin', '--monitor-plugin-whitelist', ''], config);
-			const [, usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
+			const [usedConfig,] = (application.getApplication as jest.Mock).mock.calls[0];
```
Comment (sameersubudhi) [inline test/commands/start.spec.ts:304]:
```suggestion
			const [usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
```

File: test/commands/start.spec.ts [hunk 9]
```diff
@@ -318,14 +326,14 @@ describe.skip('start', () => {
 				});
 
 			await StartCommand.run(['--config=./config.json'], config);
-			const [, usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
+			const [usedConfig,] = (application.getApplication as jest.Mock).mock.calls[0];
```
Comment (sameersubudhi) [inline test/commands/start.spec.ts:329]:
```suggestion
			const [usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
```

File: test/commands/start.spec.ts [hunk 10]
```diff
@@ -385,14 +393,14 @@ describe.skip('start', () => {
...
 			expect(usedConfig.network.seedPeers).toEqual([{ ip: 'localhost', port: 12234 }]);
 		});
 
 		it('should update the config value using env variable', async () => {
 			process.env.LISK_SEED_PEERS = 'localhost:12234,74.49.3.35:2238';
 			await StartCommand.run([], config);
-			const [, usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
+			const [usedConfig,] = (application.getApplication as jest.Mock).mock.calls[0];
```
Comment (sameersubudhi) [inline test/commands/start.spec.ts:403]:
```suggestion
			const [usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
```

File: test/commands/start.spec.ts [hunk 11]
```diff
@@ -385,14 +393,14 @@ describe.skip('start', () => {
 	describe('when seed peer is specified', () => {
 		it('should update the config value', async () => {
 			await StartCommand.run(['--seed-peers=localhost:12234'], config);
-			const [, usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
+			const [usedConfig,] = (application.getApplication as jest.Mock).mock.calls[0];
```
Comment (sameersubudhi) [inline test/commands/start.spec.ts:396]:
```suggestion
			const [usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
```

File: test/commands/start.spec.ts [hunk 12]
```diff
@@ -369,14 +377,14 @@ describe.skip('start', () => {
...
 			expect(usedConfig.network.port).toBe(1111);
 		});
 
 		it('should update the config value fro menv', async () => {
 			process.env.LISK_PORT = '1234';
 			await StartCommand.run([], config);
-			const [, usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
+			const [usedConfig,] = (application.getApplication as jest.Mock).mock.calls[0];
```
Comment (sameersubudhi) [inline test/commands/start.spec.ts:387]:
```suggestion
			const [usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
```

File: test/commands/start.spec.ts [hunk 13]
```diff
@@ -369,14 +377,14 @@ describe.skip('start', () => {
 	describe('when port is specified', () => {
 		it('should update the config value', async () => {
 			await StartCommand.run(['--port=1111'], config);
-			const [, usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
+			const [usedConfig,] = (application.getApplication as jest.Mock).mock.calls[0];
```
Comment (sameersubudhi) [inline test/commands/start.spec.ts:380]:
```suggestion
			const [usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
```

File: test/commands/start.spec.ts [hunk 14]
```diff
@@ -353,14 +361,14 @@ describe.skip('start', () => {
...
 			expect(usedConfig.logger.fileLogLevel).toBe('trace');
 		});
 
 		it('should update the config value fro menv', async () => {
 			process.env.LISK_FILE_LOG_LEVEL = 'trace';
 			await StartCommand.run([], config);
-			const [, usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
+			const [usedConfig,] = (application.getApplication as jest.Mock).mock.calls[0];
```
Comment (sameersubudhi) [inline test/commands/start.spec.ts:371]:
```suggestion
			const [usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
```

File: test/commands/start.spec.ts [hunk 15]
```diff
@@ -353,14 +361,14 @@ describe.skip('start', () => {
 	describe('when file log is specified', () => {
 		it('should update the config value', async () => {
 			await StartCommand.run(['--log=trace'], config);
-			const [, usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
+			const [usedConfig,] = (application.getApplication as jest.Mock).mock.calls[0];
```
Comment (sameersubudhi) [inline test/commands/start.spec.ts:364]:
```suggestion
			const [usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
```

File: test/commands/start.spec.ts [hunk 16]
```diff
@@ -337,14 +345,14 @@ describe.skip('start', () => {
...
 			expect(usedConfig.logger.consoleLogLevel).toBe('trace');
 		});
 
 		it('should update the config value from env', async () => {
 			process.env.LISK_CONSOLE_LOG_LEVEL = 'error';
 			await StartCommand.run([], config);
-			const [, usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
+			const [usedConfig,] = (application.getApplication as jest.Mock).mock.calls[0];
```
Comment (sameersubudhi) [inline test/commands/start.spec.ts:355]:
```suggestion
			const [usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
```

File: test/commands/start.spec.ts [hunk 17]
```diff
@@ -337,14 +345,14 @@ describe.skip('start', () => {
 	describe('when log is specified', () => {
 		it('should update the config value', async () => {
 			await StartCommand.run(['--console-log=trace'], config);
-			const [, usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
+			const [usedConfig,] = (application.getApplication as jest.Mock).mock.calls[0];
```
Comment (sameersubudhi) [inline test/commands/start.spec.ts:348]:
```suggestion
			const [usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
```

File: test/commands/start.spec.ts [hunk 18]
```diff
@@ -318,14 +326,14 @@ describe.skip('start', () => {
...
 			expect(fs.readJSON).toHaveBeenCalledWith('./config.json');
 			expect(usedConfig.logger.consoleLogLevel).toBe('error');
 		});
 
 		it('should not override the genesis config', async () => {
 			await StartCommand.run(['--config=./custom_config.json'], config);
-			const [, usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
+			const [usedConfig,] = (application.getApplication as jest.Mock).mock.calls[0];
```
Comment (sameersubudhi) [inline test/commands/start.spec.ts:336]:
```suggestion
			const [usedConfig] = (application.getApplication as jest.Mock).mock.calls[0];
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

**Review by sameersubudhi — APPROVED** [review]
LGTM 👍

**Review by ManuGowda — APPROVED** [review]

**Pull request closed without merging**